src/python/pants/backend/terraform/hcl2_parser.py

```diff
@@ -30,9 +33,14 @@ def resolve_pure_path(base: PurePath, relative_path: PurePath) -> PurePath:
 def extract_module_source_paths(path: PurePath, raw_content: bytes) -> Set[str]:
     # Import here so we can still test this file with pytest (since `hcl2` is not present in normal Pants venv.)
     import hcl2  # type: ignore[import]
+    from lark.exceptions import LarkError  # type: ignore[import]
 
     content = raw_content.decode("utf-8")
-    parsed_content = hcl2.loads(content)
+    try:
+        parsed_content = hcl2.loads(content)
+    except LarkError as e:
```

Comment:
Maybe just capture `Exception` since any exception there is problematic? Also, I would suggest not re-raising the error and instead just ignore the exception after it has been logged. This will allow Pants to proceed, but without dependency inference for the problematic file.

Then add code after https://github.com/pantsbuild/pants/blob/dc59219906f8d4dde15fa74f3acd3f36d63f8bc9/src/python/pants/backend/terraform/dependency_inference.py#L125-L131 to output any `stderr` output to the log so errors in this script are visible to Pants user.

Reply:
I don't think we should continue, because to do so would give incorrect results.

Reply:
As context: The Pants code base generally takes the stance that dependency inference failures should not cause errors. For example, see https://github.com/pantsbuild/pants/blob/b73328f4af9eea24a63b55578f6c4e5a67c3d530/src/python/pants/backend/python/dependency_inference/scripts/dependency_parser.py#L180-L183 where the Python dep inference parser will just exit "successfully" on parse errors.